src/modules/flight_mode_manager/tasks/AutoFollowTarget/follow_target_estimator/TargetEstimator.cpp

```diff
@@ -0,0 +1,358 @@
+
+	// Perform sensor fusion update if there's a new GPS message from the follow-target
+	prediction_update(deltatime);
+
+	follow_target_s follow_target;
+
+	if (_follow_target_sub.update(&follow_target)) {
+
+		// Don't perform measurement update if two follow_target messages with identical timestamps are used
+		// This can happen when using the MAVSDK and more than one outgoing follow_target message is queued.
+		const bool duplicate_measurement_received = follow_target.timestamp == _last_follow_target_timestamp;
+
+		// Skip measurements that lie in the past
+		const bool measurement_in_the_past = _last_position_fusion_timestamp >= follow_target.timestamp;
+
+		// Need at least one vehicle_local_position before estimator can work
+		const bool vehicle_local_position_invalid = _vehicle_local_position.timestamp == 0;
+
+		if (!duplicate_measurement_received && !measurement_in_the_past && !vehicle_local_position_invalid) {
+			measurement_update(follow_target);
+		}
+	}
+
+	// Keep position estimate as the last known position
+	// but stop moving the estimate
+	if (is_stale(GPS_MESSAGE_STALE_TIMEOUT_MS)) {
+		_filter_states.vel_ned_est.setZero();
+		_filter_states.acc_ned_est.setZero();
+	}
+
+	const bool states_are_finite = _filter_states.is_finite();
+
+	if (!states_are_finite) {
+		reset();
+	}
+
+	// Publish estimator message
+	follow_target_estimator_s follow_target_estimator{};
+	follow_target_estimator.timestamp = hrt_absolute_time();
+	follow_target_estimator.valid = states_are_finite;
+	follow_target_estimator.stale = is_stale(GPS_MESSAGE_STALE_TIMEOUT_MS);
+	follow_target_estimator.last_filter_reset_timestamp = _last_filter_reset_timestamp;
+	follow_target_estimator.lat_est = get_lat_lon_alt_est()(0);
+	follow_target_estimator.lon_est = get_lat_lon_alt_est()(1);
+	follow_target_estimator.alt_est = get_lat_lon_alt_est()(2);
+	_filter_states.pos_ned_est.copyTo(follow_target_estimator.pos_est);
+	_filter_states.vel_ned_est.copyTo(follow_target_estimator.vel_est);
+	_filter_states.acc_ned_est.copyTo(follow_target_estimator.acc_est);
+	follow_target_estimator.prediction_count = _prediction_count;
+	follow_target_estimator.fusion_count = _fusion_count;
+	_follow_target_estimator_pub.publish(follow_target_estimator);
+}
+
+void TargetEstimator::update_filter_gains(filter_gains_s &filter_gains) const
+{
+	const float responsiveness_param = math::constrain(_param_flw_tgt_rs.get(), .1F, 1.0F);
+
+	if (fabsf(filter_gains.responsiveness - responsiveness_param) < FLT_EPSILON) {
+		// Parameter did not change since last execution. Skip calculations
+		return;
+	}
+
+	filter_gains.responsiveness = responsiveness_param;
+
+	// The "G" gain is equivalent to "(1-responsiveness)", but beta is required for H and K gains
+	// From alpha-beta-gamma filter equations: G = 1-beta^3
+	// Therefore: beta = (1-Gp)^(1/3) = (1-(1-responsiveness))^(1/3) = (r)^(1/3)
+	const float beta_p = std::pow((filter_gains.responsiveness), 1.0f / 3.0f);
```

Comment:
This could use up quite some unexpected flash. The C library power function from NuttX would likely just be available already.